### DISCUSSION TITLE
osu! can't connect to server

### DISCUSSION THREAD

**Discussion by smpial** [osuAkatsuki/bancho.py#254] [Q&A] [answered]
in logs i found this:
`2022-07-12T18:44:15: Request to https://osu.some.domain/users () failed with RequestTimeout (retrying 1/2).`
`2022-07-12T18:44:15: Request to https://osu.some.domain/users () failed with RequestTimeout (retrying 2/2).`
`2022-07-12T18:44:15: Request to https://osu.some.domain/users () failed with RequestTimeout (FAILED).`
`2022-07-12T18:44:15: Error with account creation: System.Net.WebException: Базовое соединение закрыто: Непредвиденная ошибка при передаче. ---> System.IO.IOException: Аутентификация не пройдена из-за закрытия транспортного потока удаленной стороной.`
`2022-07-12T18:44:15: в System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)`
`2022-07-12T18:44:15: в System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)`
`2022-07-12T18:44:15: в System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)`
`2022-07-12T18:44:15: в System.Net.Security.SslState.StartSendBlob(Byte[] incoming, Int32 count, AsyncProtocolRequest asyncRequest, Boolean renegotiation)`
`2022-07-12T18:44:15: в System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest, Boolean renegotiation)`
`2022-07-12T18:44:15: в System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)`
`2022-07-12T18:44:15: в System.Net.TlsStream.CallProcessAuthentication(Object state)`
`2022-07-12T18:44:15: в System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)`
`2022-07-12T18:44:15: в System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)`
`2022-07-12T18:44:15: в System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)`
`2022-07-12T18:44:15: в System.Net.TlsStream.ProcessAuthentication(LazyAsyncResult result)`
`2022-07-12T18:44:15: в System.Net.TlsStream.Write(Byte[] buffer, Int32 offset, Int32 size)`
`2022-07-12T18:44:15: в System.Net.PooledStream.Write(Byte[] buffer, Int32 offset, Int32 size)`
`2022-07-12T18:44:15: в System.Net.ConnectStream.WriteHeaders(Boolean async)`
`2022-07-12T18:44:15: --- Конец трассировки внутреннего стека исключений ---`
but i can open all urls in browser through https and all works

**Comment by cmyui** [comment, marked as answer]
hey, seems like your osu is trying to connect to `https://some.domain` - make sure this is set to whatever your domain is in both the `-devserver` argument passed to the osu! client, as well as your nginx configuration file

**Reply by smpial** [reply]
My osu! shortcut:
C:\Users\User\AppData\Local\osu!\osu!.exe -devserver some.domain

Everything was correct in nginx, but while I was trying to solve the problem, I reinstalled it and it stopped creating the file /tmp/bancho.sock

**Reply by cmyui** [reply]
you'll need an actual domain to host the server on other than some.domain - e.g. mine is hosted at https://c.cmyui.xyz

**Reply by smpial** [reply]
I have too, but I don't want to say it, is it necessary?

**Reply by g1-1-1** [reply]
are you routing through cloudflare? try making sure your endpoints there are set correctly (especially c.[domain])

**Reply by smpial** [reply]
i have these dns records:
a.osu
assets.osu
b.osu
c.osu

do i need to make c1, c2, c3, etc.?

**Reply by g1-1-1** [reply]
i have the following:
a
api
b
c
c4
c5
c6
osu
try adding what you have missing, then see if it works

**Reply by smpial** [reply]
now i have:
a.osu
api.osu
assets.osu
b.osu
c.osu
c1.osu
c2.osu
c3.osu
c4.osu
c5.osu
c6.osu
osu.osu

but game still can't connect
`2022-07-13T07:03:10: Request to https://osu.osu.some.domain/web/osu-getseasonal.php () failed with RequestTimeout (retrying 1/2).`
`2022-07-13T07:03:10: Request to https://osu.osu.some.domain/web/osu-getseasonal.php () failed with RequestTimeout (retrying 2/2).`
`2022-07-13T07:03:10: Request to https://osu.osu.some.domain/web/osu-getseasonal.php () failed with RequestTimeout (FAILED).`
in browser all okay, and i getting an array with urls of images

**Reply by g1-1-1** [reply]
why are you using osu.[domain] as the root?

**Reply by smpial** [reply]
I use my site not only for osu, so I made sure not to get confused
and osu.[domain] worked for old server

**Reply by g1-1-1** [reply]
i'm not sure how the nginx configuration works with osu as the root domain, so you might have to ask around with people more knowledgeable about that than i am. sorry! it seems related to the root domain being osu however

**Reply by smpial** [reply]
i using from example
server_name ~^(?:c[e4]?|osu|a|b|api)\\.osu\\.some\\.domain$;

**Reply by smpial** [reply]
and there is a problem, nginx doesn't create a /tmp/bancho.sock file

**Reply by cmyui** [reply]
> and there is a problem, nginx doesn't create a /tmp/bancho.sock file

nginx.conf won't create this socket file - bancho.py will when it's started up (if configured to in the .env configuration file)

**Reply by smpial** [reply]
I fixed this and it worked!
I have a question, how to disable registration? I want to register users myself

**Reply by cmyui** [reply]
Hey apologies I must've missed that reply ^, we should likely add a feature flag for this. At the moment, you'd have to remove the register_account endpoint at the end of app/domains/osu.py
